x-pack/plugins/security_solution/server/endpoint/services/metadata/endpoint_metadata_service.ts

```diff
@@ -483,4 +488,31 @@ export class EndpointMetadataService {
       total: (docsCount as unknown as SearchTotalHits).value,
     };
   }
+
+  /**
+   * reinstalls (force delete + putTransform) all endpoint package transforms
+   * ** IMPORTANT: only intended for internal use. uses a internal saved objects client with read and update access, use with caution **
+   *
+   * @param esClient
+   * @returns installedTransforms
+   */
+  async dangerousReinstallTransforms(esClient: ElasticsearchClient) {
+    const { version } = await fetchFindLatestPackage(FLEET_ENDPOINT_PACKAGE);
+    const { packageInfo, paths } = await getRegistryPackage(FLEET_ENDPOINT_PACKAGE, version);
+    return installTransform(
+      packageInfo,
+      paths,
+      esClient,
+      createInternalReadonlySoClient(this.savedObjectsStart, new Set(['update'])),
```

Comment:
since this is called directly from the task, there is no request to create a scoped SO client. I added a new allowlist since install transform needs put access on the SO client.

Reply:
This method (to me) seems like it does not belong here. We probably should have it live somewhere else - perhaps a new service client (`EndpointSetupService`?). It is not a method that should normally be called by consumers of the endpoint metadata service IMO.

Also - why did you need to create a new internal SO client here? Was it only because of the second argument that you added to the SO client?

Reply:
if you can expose the ability to run this reinstalltransforms logic to the task without having it part of the metadata service, probably better to make it even more task-specific environment, where its less likely to be accessible to us in the handler contexts. Though this may not be strictly possible